spec/unit/puppet/provider/cs_colocation_crm_spec.rb

```diff
@@ -0,0 +1,140 @@
+require 'spec_helper'
+
+describe Puppet::Type.type(:cs_colocation).provider(:crm) do
+  before do
+    described_class.stubs(:command).with(:crm).returns 'crm'
+  end
+
+  context 'when getting instances' do
+    let :instances do
+
+      test_cib = <<-EOS
+        <cib>
+        <configuration>
+          <constraints>
+            <rsc_colocation id="first_with_second" rsc="first" score="INFINITY" with-rsc="second"/>
+          </constraints>
+        </configuration>
+        </cib>
+      EOS
+
+      described_class.expects(:block_until_ready).returns(nil)
+      if Puppet::PUPPETVERSION.to_f < 3.4
+        Puppet::Util::SUIDManager.expects(:run_and_capture).with(['crm', 'configure', 'show', 'xml']).at_least_once.returns([test_cib, 0])
+      else
+        Puppet::Util::Execution.expects(:execute).with(['crm', 'configure', 'show', 'xml']).at_least_once.returns(
+          Puppet::Util::Execution::ProcessOutput.new(test_cib, 0)
+        )
+      end
+      instances = described_class.instances
+    end
+
+    it 'should have an instance for each <colocation>' do
+      expect(instances.count).to eq(1)
+    end
+
+    describe 'each instance' do
+      let :instance do
+        instances.first
+      end
+
+      it "is a kind of #{described_class.name}" do
```

Comment:
flipping between "it should" and "it is" statements. I don't care which one you use, but stay consistent. (yes, other tests here aren't consistent either, please fix them too ;) ).